enterprise/reporting/error_reporting.go

```diff
@@ -426,9 +455,33 @@ func (edr *ErrorDetailReporter) mainLoop(ctx context.Context, c types.SyncerConf
 				errorDetailReportsDeleteQueryTimer.Since(deleteReportsStart)
 				if err != nil {
 					edr.log.Errorf("[ Error Detail Reporting ]: Error deleting local reports from %s: %v", ErrorDetailReportsTable, err)
+				} else {
+					deletedRows += len(reports)
 				}
 				// vacuum error_reports_details table
-				edr.vacuum(ctx, dbHandle, c)
+				if deletedRows >= vacuumDeletedRowThreshold.Load() {
+					if err := edr.vacuum(ctx, dbHandle, tags); err == nil {
+						deletedRows = 0
+						lastVacuum = time.Now()
+					}
+				} else {
```

Comment:
Small optimization to avoid running the query for the size estimation unless really necessary.

```suggestion
				} else if time.Since(lastVacuum) >= vacuumInterval.Load() {
```

Of course you'll have to remove the same condition from the nested `if`.